### ISSUE TITLE
when update to v1.20 got error : framework not found flutter_plugin_android_lifecycle

### ISSUE THREAD

**Issue by yaseen-abugamess** [flutter/flutter#63220]
Launching lib/main.dart on iPhone 11 in debug mode...
Running pod install...
Running Xcode build...
Xcode build done.                                           58.5s
Failed to build iOS app
Could not build the application for the simulator.
Error launching application on iPhone 11.
Error output from Xcode build:
↳
** BUILD FAILED **

Xcode's output:
↳
ld: framework not found flutter_plugin_android_lifecycle
clang: error: linker command failed with exit code 1 (use -v to see invocation)
note: Using new build system
note: Building targets in parallel
note: Planning build
note: Constructing build description

**Comment by stx** [comment]
Same problem.

**Solution:** Remove this from *Xcode > Targets > Runner > Other Linker Flags:*

    -framework
    flutter_plugin_android_lifecycle

No idea if this is a good idea or not, but Flutter won't build otherwise. Is **Other Linker Flags** safe to modify? What builds the list?

**Comment by ikarelin** [comment]
Same problem. Do you think removing the linker flag will not affect application running ?

**Comment by yaseen-abugamess** [comment]
Sure, it has an effect, but I'm back to the older version 1.7.5 and work normally until the problem is resolved

> Same problem. Do you think removing the linker flag will not affect application running ?



> Same problem.
> 
> **Solution:** Remove this from _Xcode > Targets > Runner > Other Linker Flags:_
> 
> ```
> -framework
> flutter_plugin_android_lifecycle
> ```
> 
> Ran into this problem with _GoogleDataTransportCCTSupport_ as well as seen in [FirebaseExtended/flutterfire#3078](https://github.com/FirebaseExtended/flutterfire/issues/3078).
> 
> No idea if this is a good idea or not, but Flutter won't build otherwise.
> 
> Is **Other Linker Flags** safe to modify? What builds the list?



> Same problem.
> 
> **Solution:** Remove this from _Xcode > Targets > Runner > Other Linker Flags:_
> 
> ```
> -framework
> flutter_plugin_android_lifecycle
> ```
> 
> Ran into this problem with _GoogleDataTransportCCTSupport_ as well as seen in [FirebaseExtended/flutterfire#3078](https://github.com/FirebaseExtended/flutterfire/issues/3078).
> 
> No idea if this is a good idea or not, but Flutter won't build otherwise.
> 
> Is **Other Linker Flags** safe to modify? What builds the list?

Sure, it has an effect, but I'm back to the old version v1.7.5 and work normally until the problem is resolved

**Comment by jersocOD** [comment]
Hello guys
I had that error even after switching to v1.7.5,
I solved it with these commands:
Before running, make a backup
```
flutter clean 
rm ios/Podfile
rm ios/Podfile.lock 
rm pubspec.lock 
rm -rf ios/Pods ios/Runner.xcworkspace 
flutter run
```
Reference: @https://github.com/flutter/flutter/issues/41900#issuecomment-601599410

**Comment by KohlsAdrian** [comment]
> Hello guys
> I had that error even after switching to v1.7.5,
> I solved it with these commands:
> Before running, make a backup
> 
> ```
> flutter clean 
> rm ios/Podfile
> rm ios/Podfile.lock 
> rm pubspec.lock 
> rm -rf ios/Pods ios/Runner.xcworkspace 
> flutter run
> ```
> 
> Reference: @[#41900 (comment)](https://github.com/flutter/flutter/issues/41900#issuecomment-601599410)

Does not work

**Comment by moseskarunia** [comment]
This is the only red message when running --verbose

**Comment by markusaksli-nc** [comment]
Hi @yaseen-abugamess 
Could you please provide your `pubspec.yaml` and `flutter doctor -v` (before and after upgrading)? It would be very helpful if any of you running into this issue could provide detailed steps on how to reproduce the issue or a minimal reproduction project repository that demonstrates this issue in isolation without any production code or third-party plugins. We need to know what **exactly** is causing the issue here.

**Comment by stx** [comment]
In at least some previous Flutter projects, for whatever reason, `Other Linker Flags` is populated by a ton of dependencies. In new projects, it is just `$(inherited) -framework Flutter`.

Here's a pre-1.20 project snippet from `ios/Runner.xcodeproj/project.pbxproj`. This section is completely missing on a 1.20 project, even after compiling. Not sure where these flags were populated, but appears to be the issue.

`	OTHER_LDFLAGS = (
					"$(inherited)",
					"-ObjC",
					"-l\"c++\"",
					"-l\"sqlite3\"",
					"-l\"z\"",
					"-framework",
					"\"Accelerate\"",
					"-framework",
					"\"Alamofire\"",
					"-framework",
					"\"CFNetwork\"",
					"-framework",
					"\"CoreData\"",
					"-framework",
					"\"CoreGraphics\"",
					"-framework",
					"\"CoreImage\"",
					"-framework",
					"\"CoreLocation\"",
					"-framework",
					"\"CoreTelephony\"",
					"-framework",
					"\"CoreText\"",
					"-framework",
					"\"Crashlytics\"",
					"-framework",
					"\"FBLPromises\"",
					"-framework",
					"\"FIRAnalyticsConnector\"",
					"-framework",
					"\"Fabric\"",
					"-framework",
					"\"FirebaseABTesting\"",
					"-framework",
					"\"FirebaseAnalytics\"",
					"-framework",
					"\"FirebaseAuth\"",
					"-framework",
					"\"FirebaseCore\"",
					"-framework",
					"\"FirebaseCoreDiagnostics\"",
					"-framework",
					"\"FirebaseFirestore\"",
					"-framework",
					"\"FirebaseFunctions\"",
					"-framework",
					"\"FirebaseInstallations\"",
					"-framework",
					"\"FirebaseInstanceID\"",
					"-framework",
					"\"FirebaseMessaging\"",
					"-framework",
					"\"FirebasePerformance\"",
					"-framework",
					"\"FirebaseRemoteConfig\"",
					"-framework",
					"\"Flutter\"",
					"-framework",
					"\"Foundation\"",
					"-framework",
					"\"GLKit\"",
					"-framework",
					"\"GTMSessionFetcher\"",
					"-framework",
					"\"GoogleAppMeasurement\"",
					"-framework",
					"\"GoogleDataTransport\"",
					"-framework",
					"\"GoogleMaps\"",
					"-framework",
					"\"GoogleMapsBase\"",
					"-framework",
					"\"GoogleMapsCore\"",
					"-framework",
					"\"GoogleToolboxForMac\"",
					"-framework",
					"\"GoogleUtilities\"",
					"-framework",
					"\"ImageIO\"",
					"-framework",
					"\"KeychainAccess\"",
					"-framework",
					"\"Metal\"",
					"-framework",
					"\"MobileCoreServices\"",
					"-framework",
					"\"OpenGLES\"",
					"-framework",
					"\"QuartzCore\"",
					"-framework",
					"\"SafariServices\"",
					"-framework",
					"\"Security\"",
					"-framework",
					"\"StoreKit\"",
					"-framework",
					"\"SwiftyJSON\"",
					"-framework",
					"\"SystemConfiguration\"",
					"-framework",
					"\"UIKit\"",
					"-framework",
					"\"absl\"",
					"-framework",
					"\"cloud_firestore\"",
					"-framework",
					"\"cloud_functions\"",
					"-framework",
					"\"firebase_auth\"",
					"-framework",
					"\"firebase_core\"",
					"-framework",
					"\"firebase_crashlytics\"",
					"-framework",
					"\"firebase_messaging\"",
					"-framework",
					"\"firebase_performance\"",
					"-framework",
					"\"flutter_secure_storage\"",
					"-framework",
					"\"google_maps_flutter\"",
					"-framework",
					"\"grpc\"",
					"-framework",
					"\"grpcpp\"",
					"-framework",
					"\"leveldb\"",
					"-framework",
					"\"local_auth\"",
					"-framework",
					"\"location\"",
					"-framework",
					"\"nanopb\"",
					"-framework",
					"\"openssl_grpc\"",
					"-framework",
					"\"protobuf\"",
					"-framework",
					"\"shared_preferences\"",
					"-framework",
					"\"url_launcher\"",
					"-framework",
					"\"vibrate\"",
					"-weak_framework",
					"\"UserNotifications\"",
					"$(inherited)",
					"-framework",
					Flutter,
				);`

**Comment by jersocOD** [comment]
Any ideas here? #61063

**Comment by moseskarunia** [comment]
I wrote an article on dev.to because I finally succeed to debug in physical device on 1.20.2

https://dev.to/moseskarunia/fixing-flutter-ios-build-fail-2mcm

Hope it helps somebody out there!

**Comment by ollydixon** [comment]
@stx it resets on clean build :-/

**Comment by darshankawar** [comment]
Without additional information, we are unfortunately not sure how to resolve this issue. 
We are therefore reluctantly going to close this bug for now. 
Please don't hesitate to comment on the bug if you have any more information for us; we will reopen it right away!  
Thanks for your contribution.

Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of `flutter doctor -v`.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.

**Comment by delizondo** [comment]
Got the same problem, and was able to resolve it. Not entirely happy with the process, but at least it work.

1. Backup the whole ios folder
2. Delete the ios folder
3. in the project root `flutter create .`
4. Restore any configuration you previously had in the iOS Project
5. Run the project.

**Comment by ollydixon** [comment]
@delizondo any ideas to fix this on a build server? Version 1.2.N has been a nightmare, one of their worse updates to date.

**Comment by ollydixon** [comment]
@darshankawar the issue is kinda obvious. I don't think it should have been closed. Still an issue that effects all the build servers.

**Comment by darshankawar** [comment]
@ollydixon I understand. Usually when we add `wait` label and if OP doesn't respond in 21 days, we go ahead and close the issue. As a closing note, we mention:

```
Could everyone who still has this problem please file a new issue with the exact description of what happens, logs, and the output of flutter doctor -v.
All system setups can be slightly different, so it's always better to open new issues and reference related issues.
```

**Comment by ollydixon** [comment]
@darshankawar but the issue is straightforwards, ```flutter_plugin_android_lifecycle``` is added to ```framework``` each time ```flutter clean``` happens. It doesn't matter what system you are running, iOS, Flutter version ```1.20.X```. You can test this now; it's a major issue for anyone doing flutter processionally with build servers.

**Comment by KohlsAdrian** [comment]
> @darshankawar but the issue is straightforwards, `flutter_plugin_android_lifecycle` is added to `framework` each time `flutter clean` happens. It doesn't matter what system you are running, iOS, Flutter version `1.20.X`. You can test this now; it's a major issue for anyone doing flutter processionally with build servers.

This is the answer I wanted to make, but didn't know the correct words and matches. This is the most straight forwarded issue that is happening.

**Comment by delizondo** [comment]
@ollydixon no idea how to fix this in a build server, I fix the problem locally in my machine. However, something I did notice when doing a diff comparison in the `project.pbxproj` file is that the same chunk of code that @stx mention in his [response](https://github.com/flutter/flutter/issues/63220#issuecomment-671626129) was removed.

**Comment by ZeroCool2u** [comment]
Running into the same issue on local OS X/iOS builds that were upgraded from 1.X.

**Comment by KohlsAdrian** [comment]
Try [this](https://github.com/flutter/flutter/issues/60883#issuecomment-654602918)

**Comment by AustrisC** [comment]
After longer investigation, I identified **two error-causing points**.

### Firstly,
Changing linker flags in `ios/Runner.xcodeproj/project.pbxproj `.
In Xcode, navigate to `Project Runner` -> `Build Settings` -> `Other Linker Flags` & change all the blob that there with:
```
$(inherited)
-framework
Flutter
```
<img width="857" alt="linker-flags" src="https://user-images.githubusercontent.com/12381652/94800132-d70c3a00-03ec-11eb-9083-8820149b540f.png">

### Secondly,
VS Code entry in my `PATH` variable was incorrect. After changing it accordingly to the documentation [here](https://code.visualstudio.com/docs/setup/mac) he error was no longer present.
